File: 2025-07-July-GFG-POTD/July 28 - Make Matrix Beautiful/Solution.py
class Solution:
    @staticmethod
    def balanceSums(mat):
        n = len(mat)
        row_sum = [0] * n
        col_sum = [0] * n
        max_sum = 0

        # Calculate row and column sums
        for i in range(n):
            for j in range(n):
                row_sum[i] += mat[i][j]
                col_sum[j] += mat[i][j]

        # Determine the target sum (maximum row or column sum)
        for i in range(n):
            max_sum = max(max_sum, row_sum[i], col_sum[i])

        # Calculate number of operations needed
        operations = 0
        for i in range(n):
            operations += (max_sum - row_sum[i])

        return operations

File: 2025-07-July-GFG-POTD/July 28 - Make Matrix Beautiful/Solution.cpp
class Solution {
public:
    static int balanceSums(vector<vector<int>>& mat) {
        int n = mat.size();
        vector<int> rowSum(n, 0), colSum(n, 0);
        int maxSum = 0;

        // Calculate row and column sums
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n; ++j) {
                rowSum[i] += mat[i][j];
                colSum[j] += mat[i][j];
            }
        }

        // Determine the target sum (maximum row or column sum)
        for (int i = 0; i < n; ++i) {
            maxSum = max(maxSum, rowSum[i]);
            maxSum = max(maxSum, colSum[i]);
        }

        // Calculate number of operations needed
        int operations = 0;
        for (int i = 0; i < n; ++i) {
            operations += (maxSum - rowSum[i]);
        }

        return operations;
    }
};
